rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -0,0 +1,156 @@
+               FROM participant_summary ps
+              WHERE %(filters)s
+           GROUP BY day) member
+             ON c2.day = member.day
+           LEFT OUTER JOIN
+            (SELECT COUNT(*) cnt,
+             DATE(CASE WHEN enrollment_status = 3 THEN
+                   GREATEST(consent_for_electronic_health_records,
+                            questionnaire_on_the_basics_time,
+                            questionnaire_on_lifestyle_time,
+                            questionnaire_on_overall_health_time,
+                            physical_measurements_time,
+                            CASE WHEN sample_status_1ed04_time IS NOT NULL
+                             THEN
+                             (CASE WHEN sample_status_1sal_time IS NOT
+                                NULL
+                                THEN LEAST(sample_status_1ed04_time,
+                                           sample_status_1sal_time)
+                                ELSE sample_status_1ed04_time END)
+                             ELSE sample_status_1sal_time END)
+                   ELSE NULL END) day
+               FROM participant_summary ps
+              WHERE %(filters)s
+           GROUP BY day) full
+             ON c2.day = full.day) day_sums, calendar
+          WHERE calendar.day >= :start_date
+            AND calendar.day <= :end_date
+          GROUP BY calendar.day
+          ORDER BY calendar.day;
+      """ % {'filters': filters_sql}
+
+    params = {'start_date': start_date, 'end_date': end_date}
+
+    results_by_date = []
+
+    with self.session() as session:
+      cursor = session.execute(sql, params)
+
+    # Iterate through each result (by date), transforming tabular SQL results
+    # into expected list-of-dictionaries response format
+    try:
+      results = cursor.fetchall()
+      for result in results:
+        date = result[-1]
+        metrics = {}
+        values = result[:-1]
+        for i, value in enumerate(values):
+          key = strata[i]
+          if value == None:
+            value = 0
+          metrics[key] = int(value)
+        results_by_date.append({
+          'date': str(date),
+          'metrics': metrics
+        })
+    finally:
+      cursor.close()
+
+    return results_by_date
+
+
+  def get_facets_sql(self, facets):
+    """Helper function to transform facets/filters selection into SQL
+
+    :param facets: Object representing facets and filters to apply to query results
+    :return: SQL for 'WHERE' clause, reflecting filters specified in UI
+    """
+
+    facets_sql = []
+
+    facet_map = {
+      'awardee_ids': 'hpo_id',
+      'enrollment_statuses': 'enrollment_status'
+    }
+
+    for facet in facets:
+      filters_sql = []
+      db_field = facet_map[facet]
+      filters = facets[facet]
+      for q_filter in filters:
+        if str(q_filter) != '':
+          filters_sql.append('ps.' + db_field + ' = ' + str(int(q_filter)))
```

Comment:
Seems like when there are multiple values for a facet, we should use an in clause instead of many ORs?

If you are going to use ORs, you'll want to put parens around the OR expressions, or order of operations may be wrong.

Reply:
Also: we could consider using bound parameters here instead of inlining the values in the SQL. We do that in other places using this function:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/database_utils.py#L16

This may help the SQL perform slightly better since the execution plan for the query can be cached when the only thing changing are the bound params.

Reply:
> If you are going to use ORs, you'll want to put parens around the OR expressions, or order of operations may be wrong.

Done, thanks for the catch.

Reply:
I think the SQL for using an IN clause with bound parameters would be simpler than this, FWIW. But this works.

Reply:
Agreed.  I've noted the IN clause refactoring as a possible future task.  For now, given time constraints, since this works and isn't especially complex, I propose we use what's implemented here.